src/app/public/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.component.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  OnInit,
+  ViewChild
+} from '@angular/core';
+
+import {
+  SkyLibResourcesService
+} from '@skyux/i18n';
+
+import {
+  ICellEditorAngularComp
+} from 'ag-grid-angular';
+
+import {
+  ICellEditorParams
+} from 'ag-grid-community';
+
+@Component({
+  selector: 'sky-ag-grid-cell-editor-autocomplete',
+  templateUrl: './cell-editor-autocomplete.component.html',
+  styleUrls: ['./cell-editor-autocomplete.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyAgGridCellEditorAutocompleteComponent implements ICellEditorAngularComp, OnInit {
+  public currentSelection: any;
+  public autocompleteInputLabel: string;
+
+  public data: any[] = [];
+  public debounceTime: number;
+  public descriptorProperty: string;
+  public propertiesToSearch: string[];
+  public search: (searchText: string, data?: any[]) => any[] | Promise<any[]>;
+  public searchFilters: (searchText: string, item: any) => boolean;
+  public searchResultsLimit: number;
+  public searchResultTemplate: string;
+  public searchTextMinimumCharacters: number;
+  public selectionChange: Function;
+
+  public columnWidth: number;
+  public rowHeight: number;
+  public columnHeader: string;
+  public rowNumber: number;
+  private params: ICellEditorParams;
+
+  @ViewChild('skyCellEditorAutocomplete', {read: ElementRef})
+  public input: ElementRef;
+
+  constructor(private libResources: SkyLibResourcesService) { }
+
+  public agInit(params: ICellEditorParams) {
+    this.params = params;
+    this.currentSelection = this.params.value;
```

Comment:
Should this code be more defensive, i.e. is it possible that something like `this.params.column` might be undefined, causing an error when calling `getActualWidth()`? I know this function is called by ag-Grid directly, so if they guarantee these properties will be set, then there's no change needed here.

Reply:
my understanding is that they should always be set, but I'm not seeing that explicitly stated anywhere so I will go ahead and make it more defensive to be safe